api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/FhirObservationDaoImpl.java

```diff
@@ -160,6 +169,27 @@ private void handleValueCodedConcept(Criteria criteria, TokenAndListParam valueC
 		}
 	}
 	
+	private Boolean validReferenceParam(ReferenceParam ref) {
+		return (ref != null && ref.getIdPart() != null && ref.getResourceType() != null);
+	}
+	
```

Comment:
Is there any value to having this as a function? Also, if we leave the function in here, the return type should be `boolean` rather than `Boolean` (it's usually better to use primitives where possible).